Turn 'deployment considerations' into 'deployment path' (#3456)
theforeman/foreman-documentation: guides/common/modules/con_creating-and-managing-roles.adoc

[id="Creating_and_Managing_Roles_{context}"]
ifeval::["{context}" == "admin"]
= Creating and managing roles
endif::[]
ifeval::["{context}" == "planning"]
= Defining role-based access control policies
endif::[]

> Users in {Project} can have one or more roles attached.

Reviewer: ```suggestion
Users in {Project} can have one or more roles assigned.
```

These roles are associated with permissions that enable users to perform specified administrative actions in {Project}.
Permission filters define the actions allowed for a certain resource type.

{Project} provides a set of predefined roles with permissions sufficient for standard tasks.
ifeval::["{context}" == "admin"]
For a list of these roles, see xref:Predefined_Roles_Available_in_{project-context}_{context}[].
endif::[]
You can also configure custom roles.

Apart from the administrator role, the following types of roles are common:

> Roles related to applications or parts of infrastructure:: For example, roles for owners of {RHEL} as the operating system versus owners of application servers and database servers.

Reviewer 2: ```suggestion
Roles related to applications or parts of infrastructure:: For example, roles for owners of {client-os} as the operating system versus owners of application servers and database servers.
```

Reviewer 2: please also split the line at "::".

Author: Addressed.

> Roles related to a particular stage of the software lifecycle:: For example, roles divided among the development, testing, and production phases, where each phase has one or more owners.

Reviewer 2: ```suggestion
Roles related to a particular stage of the software lifecycle::
For example, roles divided among the development, testing, and production phases, where each phase has one or more owners.
```

> Roles related to specific tasks:: For example, role of a security manager or role of a license manager.

Reviewer 2: ```suggestion
Roles related to specific tasks::
For example, role of a security manager or role of a license manager.
```

ifeval::["{context}" == "planning"]
.Additional resources

> * For more information, see link:{AdministeringDocURL}Managing_Users_and_Roles_admin[_{AdministeringDocTitle}_].

Reviewer 2: ```suggestion
* For more information, see {AdministeringDocURL}Managing_Users_and_Roles_admin[_{AdministeringDocTitle}_].
```

[id="Creating_and_Managing_Roles_{context}"]
ifeval::["{context}" == "admin"]
= Creating and managing roles
endif::[]
ifeval::["{context}" == "planning"]

> = Defining role-based access control policies

Author: It might be good to mention this somewhere around here:

* https://community.theforeman.org/t/organizing-access-in-foreman-seems-overly-tedious/40278/8
* https://github.com/theforeman/foreman-documentation/blob/e02c19fe9ab38bd94bde08d450134aa73c228ead/guides/common/modules/ref_predefined-roles-available-in-project.adoc?plain=1#L24

[id="Creating_and_Managing_Roles_{context}"]
ifeval::["{context}" == "admin"]

> = Creating and managing roles

Reviewer 2: note to @maximiliankolb Check if this works downstream.

{Project} provides a set of predefined roles with permissions sufficient for standard tasks.
ifeval::["{context}" == "admin"]
For a list of these roles, see xref:Predefined_Roles_Available_in_{project-context}_{context}[].
endif::[]
You can also configure custom roles.

[NOTE]
====
One of the predefined roles is the *Default role*.
{Project} assigns the *Default role* to every user in the system.
By default, the *Default role* grants only a limited set of permissions.
Be aware that if you add a permission to the *Default role*, every {Project} users will gain that permission.
Assigning a different role to a user does not remove the *Default role* from the user.
====

The following types of roles are commonly defined within various {Project} deployments:

Roles related to applications or parts of infrastructure::
For example, roles for owners of {client-os} as the operating system as opposed to roles for owners of application servers and database servers.

Roles related to a particular stage of the software lifecycle::
For example, roles divided among the development, testing, and production phases, where each phase has one or more owners.

Roles related to specific tasks::

> For example, role of a security manager or role of a license manager.

Reviewer 3: _-Wpedantic_

> role of a license manager

If someone came to us and asked "I'm a license manager, the docs say a role can be created for me. How do I set it up?", would we be able to answer that?

For example, roles divided among the development, testing, and production phases, where each phase has one or more owners.

Roles related to specific tasks::

> For example, you can create a role for security managers and a role for license managers, depending on the specific tasks employees need to be able to perform within your organization.

Reviewer 2: ```suggestion
For example, you can create a role for security managers and a role for license managers, depending on the specific tasks users need to be able to perform within your organization.
```
